packages/frontend/workspace/src/list/cache.ts

```diff
@@ -0,0 +1,23 @@
+import { type WorkspaceMetadata } from '../metadata';
+
+const CACHE_STORAGE_KEY = 'jotai-workspaces';
+
+export function readWorkspaceListCache() {
+  const metadata = localStorage.getItem(CACHE_STORAGE_KEY);
+  if (metadata) {
+    try {
+      const items = JSON.parse(metadata) as WorkspaceMetadata[];
+      return [...items];
+    } catch (e) {
+      console.error('cannot parse worksapce', e);
+    }
+    return [];
+  }
+  return [];
+}
+
+export function writeWorkspaceListCache(metadata: WorkspaceMetadata[]) {
+  const metadataMap = new Map(metadata.map(x => [x.id, x]));
+  metadata = Array.from(metadataMap.values());
```

Comment:
why trasform the data back and forth using a map? to dedupe the values by id?

Reply:
This is copy from old code
now it's removed